### PR TITLE
DM-28947: Add option to optionally set spectra init

### PR DESCRIPTION
**Pull request by fred3m**

### CODE REVIEW

File: python/lsst/meas/extensions/scarlet/scarletDeblendTask.py [hunk 1]
```diff
@@ -240,6 +240,15 @@ def deblend(mExposure, footprint, config):
...
     centers = [np.array([peak.getIy()-ymin, peak.getIx()-xmin], dtype=int) for peak in footprint.peaks]
 
+    # Choose whether or not to use the improved spectral initialization
+    if config.setSpectra:
+        if config.maxSpectrumCutoff == -1:
+            spectrumInit = True
+        else:
+            spectrumInit = len(centers) * bbox.getArea() < config.maxSpectrumCutoff
+    else:
+        spectrumInit = config.setSpectra
```
Comment (erykoff) [inline python/lsst/meas/extensions/scarlet/scarletDeblendTask.py]: I think in this case this is always `False`, right?  I think it should just be set to `False`.

File: python/lsst/meas/extensions/scarlet/scarletDeblendTask.py [hunk 2]
```diff
@@ -240,6 +240,15 @@ def deblend(mExposure, footprint, config):
     ymin = bbox.getMinY()
     centers = [np.array([peak.getIy()-ymin, peak.getIx()-xmin], dtype=int) for peak in footprint.peaks]
 
+    # Choose whether or not to use the improved spectral initialization
+    if config.setSpectra:
+        if config.maxSpectrumCutoff == -1:
```
Comment (erykoff) [inline python/lsst/meas/extensions/scarlet/scarletDeblendTask.py]: Would it be simpler here (and below) to say that anything `<= 0` turns this off?

**Review by erykoff — APPROVED** [review]
A couple of minor comments that I think will make it a bit easier to read/use.

File: python/lsst/meas/extensions/scarlet/scarletDeblendTask.py (hunk 1 above)
Comment (fred3m) [reply python/lsst/meas/extensions/scarlet/scarletDeblendTask.py]: Yes. Originally I tried to use `True` to always use `setSpectra`, `False` to never use it, and `None` to check `maxSpectrumCutoff`, but the stack won't let me use `None` as the default for a `bool` (which makes sense, but :( ).

**Comment by erykoff** [comment]
One more thing: the title of the PR has a typo.

**Pull request merged**